Lib/_pyio.py

```diff
@@ -123,10 +123,11 @@ def open(file, mode="r", buffering=-1, encoding=None, errors=None,
     the size of a fixed-size chunk buffer.  When no buffering argument is
     given, the default buffering policy works as follows:
 
-    * Binary files are buffered in fixed-size chunks; the size of the buffer
-      is chosen using a heuristic trying to determine the underlying device's
-      "block size" and falling back on `io.DEFAULT_BUFFER_SIZE`.
-      On many systems, the buffer will typically be 4096 or 8192 bytes long.
+   * Binary files are buffered in fixed-size chunks; the size of the buffer
+     is set to `max(io.DEFAULT_BUFFER_SIZE, st_blksize)` using a heuristic
```

Comment:
`st_blksize` here is used before it's ever defined, how about something like:

Binary files are buffered in fixed-size chunks; the size of the buffer is the max of the default block size defaults to the file's block size. On most systems, the buffer will typically be 131072 bytes long.